sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/GroupByKeyTest.java

```diff
@@ -173,6 +175,74 @@ public Void apply(Iterable<KV<String, Iterable<Integer>>> actual) {
   }
 
   @Test
+  @Category(ValidatesRunner.class) public void testGroupByKeyMultipleWindows() {
+    PCollection<KV<String, Integer>> windowedInput =
+        p.apply(
+                Create.timestamped(
+                    TimestampedValue.of(KV.of("foo", 1), new Instant(1)),
+                    TimestampedValue.of(KV.of("foo", 4), new Instant(4)),
+                    TimestampedValue.of(KV.of("bar", 3), new Instant(3))))
+            .apply(
+                Window.<KV<String, Integer>>into(
+                    SlidingWindows.of(Duration.millis(5L)).every(Duration.millis(3L))));
+
+    PCollection<KV<String, Iterable<Integer>>> output =
+        windowedInput.apply(GroupByKey.<String, Integer>create());
+
+    PAssert.that(output)
+        .satisfies(
```

Comment:
Can we `PAssert.that(output).inWindow(...).containsInAnyOrder(...)` to tighten up the test slightly?